### PR TITLE
Ruby

### PR DESCRIPTION
**Pull request by anup-y**
Exercise Completed

### CODE REVIEW

File: RUBY/exercise01/Calculations.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Calculations
+@@string =""
+@@finalString =""
+    def initialize(string)
+        @@string = string
+    end
+
+    def average_calculation(string)
+        a,b, c = string.split('-')
+        b = b.to_f
```
Comment (minaxi98) [inline RUBY/exercise01/Calculations.rb]: Change variables name, Why class variables needed.

File: RUBY/exercise01/unitTest.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'test/unit'
+require_relative 'Calculations'
+
+class TestAverage < Test::Unit::TestCase
+    def test_make_calculations_object
```
Comment (minaxi98) [inline RUBY/exercise01/unitTest.rb:5]: Handle cases when invalid input in entered

File: RUBY/exercise02/exercise_two.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+class Array
+# @@arr
+#     def initialize(arr)
+#         @@arr = arr
+#     end
+
+    def element_at(arr,index)
```
Comment (minaxi98) [inline RUBY/exercise02/exercise_two.rb:9]: Handle array index out of bound

File: RUBY/exercise02/exercise_two.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    def inclusive_range(arr,start_pos,end_pos)
+        return arr[start_pos..end_pos]
+    end
+
+    def non_inclusive_range(arr,start_pos,end_pos)
+        return arr[start_pos...end_pos]
+    end
+
+    def start_and_length(arr,start_pos,length)
+        return length-start_pos
```
Comment (minaxi98) [inline RUBY/exercise02/exercise_two.rb]: Check the question, not expected solution

File: RUBY/exercise02/unitTest.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    end
+
+    def test_non_inclusive_range
+        test_non_inclusive = Array.new
+        assert_equal([1,2],test_non_inclusive.non_inclusive_range([9,5,1,2,3,4,0,-1],2,4))
+    end
+
+    def test_start_length
+        test_start_and = Array.new
+        assert_equal(7,test_start_and.start_and_length([9,5,1,2,3,4,0,-1],1,8))
```
Comment (minaxi98) [inline RUBY/exercise02/unitTest.rb]: test cases for edge cases

File: RUBY/exercise03/exercise_three.rb [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            if(!key.is_a?Integer)
+                hash_values.delete(key)
+            end
+        end
+        # printHash(hash_values)
+    end
+
+    def delete_even_keys(hash_values)
+        for key,value in hash_values do
+            if (key % 2 == 0)
```
Comment (minaxi98) [inline RUBY/exercise03/exercise_three.rb]: can be done in function

File: RUBY/exercise03/unitTest.rb [hunk 7]
```diff
@@ -0,0 +1,77 @@
+require 'test/unit'
+require_relative 'exercise_three'
+
+class TestAverage < Test::Unit::TestCase
+   
+    def test_hash_creation
+        test_hash_creation = Hashoperations.new({
```
Comment (minaxi98) [inline RUBY/exercise03/unitTest.rb:7]: test case with string key can be added

File: RUBY/exercise04/exercise_four.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+    def initialize(sports,skip_num)
+        @@sports = sports
+        @@skip_num = skip_num
+    end
+
+    def skip_sports(sports,skip_num)
+        sports.each_with_index do |element, index|
+            next if index < skip_num
+            print index,":",element," "
```
Comment (minaxi98) [inline RUBY/exercise04/exercise_four.rb:23]: this should be stored in array

File: RUBY/exercise04/unitTest.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+class TestAverage < Test::Unit::TestCase
+
+    def test_create_sports
+        test_sports_creation = Sports.new(['Criket','Football','Tennis','Hockey'],2)
+        assert test_sports_creation.instance_of? Sports
+    end
+
+    def test_skip_sports
+        test_sports_creation = Sports.new(['Criket','Football','Tennis','Hockey'],2)
```
Comment (minaxi98) [inline RUBY/exercise04/unitTest.rb:12]: test case for edge case

File: RUBY/exercise05/exercise_five.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        end
+    end
+
+    def write_to_file
+        file_path = "Anup_#{Time.now.utc.to_i}.json"
+        # File.new(file_path,'w')# print file_path
+        File.open(file_path,'w') do |file|
+            file.write(JSON.dump(@@data))
+        end
+        if(!is_empty = File.size(file_path) == 0)
```
Comment (minaxi98) [inline RUBY/exercise05/exercise_five.rb:34]: instead of return 1 u can use proper string like "File write successful".
 Not correct utc format

File: RUBY/exercise05/exercise_five.rb [hunk 11]
```diff
@@ -0,0 +1,39 @@
+require 'json'
+
+class Jsonoperations
+@@data =''
+@@file_path =''
+    def initialize(data,file_path)
```
Comment (minaxi98) [inline RUBY/exercise05/exercise_five.rb]: No need to declare class variables

File: RUBY/exercise06/exercise_six.rb [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+        # puts matches
+        @@encoded_string = Base64.encode64(matches)
+        # puts encoded_string
+        if(matches)
+            return 1
+        end
+    end
+
+    def write_file
+        data=File.read(@@file_path)
```
Comment (minaxi98) [inline RUBY/exercise06/exercise_six.rb]: use exception handling here

**Review by minaxi98 — CHANGES_REQUESTED** [review]
